src/main/scala/de/heikoseeberger/sbtheader/HeaderPlugin.scala

```diff
@@ -32,7 +33,7 @@ object HeaderPattern {
   def commentBetween(start: String, middle: String, end: String): Regex =
     new Regex(raw"""(?s)($start(?!\$middle).*?$end(?:\n|\r|\r\n)+)(.*)""")
   def commentStartingWith(start: String): Regex =
-    new Regex(raw"""(?s)((?:$start[^\n\r]*(?:\n|\r|\r\n))+(?:\n|\r|\r\n)+)(.*)""")
+    new Regex(raw"""(?s)((?:$start[^\n\r]*(?:\n|\r|\r\n))?(?:#[^\n\r]*(?:(?:\n){2,}|(?:\r){2,}|(?:\r\n){2,})+))(.*)""")
 }
```

Comment:
@hseeberger I have problems passing some tests with original regex. This new regex should fix the problem. Look at this simplified examples to see the problem:
https://regex101.com/r/sF6fB7/1 
https://regex101.com/r/rB4wD3/1